pkg/db/k8sdetail.go

```diff
@@ -0,0 +1,67 @@
+package db
+
+import (
+	"encoding/json"
+	"github.com/aquasecurity/trivy-db/pkg/types"
+	bolt "go.etcd.io/bbolt"
+	"golang.org/x/xerrors"
+)
+
+const k8sRootBucket = "k8s"
+const DbBucket = "db"
+const DataSourceBucket = "data-source"
+
+func (dbc Config) PutK8sDataSource(tx *bolt.Tx, bktName string, source types.DataSource) (err error) {
+	rootBucket, err := tx.CreateBucketIfNotExists([]byte(k8sRootBucket))
+	if err != nil {
+		return xerrors.Errorf("failed to create %s bucket: %w", k8sRootBucket, err)
+	}
+	dataSource, err := rootBucket.CreateBucketIfNotExists([]byte(DataSourceBucket))
+	if err != nil {
+		return xerrors.Errorf("failed to create %s bucket: %w", DataSourceBucket, err)
+	}
+	value, err := json.Marshal(source)
+	if err != nil {
+		return xerrors.Errorf("JSON marshal error: %w", err)
+	}
+
+	return dataSource.Put([]byte(bktName), value)
+}
+
+func (dbc Config) PutK8sDb(tx *bolt.Tx, key string, apis interface{}) (err error) {
+	rootBucket, err := tx.CreateBucketIfNotExists([]byte(k8sRootBucket))
+	if err != nil {
+		return xerrors.Errorf("failed to create %s bucket: %w", k8sRootBucket, err)
+	}
+	dbBucket, err := rootBucket.CreateBucketIfNotExists([]byte(DbBucket))
+	if err != nil {
+		return xerrors.Errorf("failed to create %s bucket: %w", DbBucket, err)
+	}
+	value, err := json.Marshal(apis)
+	if err != nil {
+		return xerrors.Errorf("JSON marshal error: %w", err)
+	}
+	if err := dbBucket.Put([]byte(key), value); err != nil {
+		return xerrors.Errorf("failed to put k8s api data: %w", err)
+	}
+	return nil
+}
+
+func (dbc Config) GetK8sDb(key string) (interface{}, error) {
+	var outdatedapi types.OutDatedAPIData
+	err := db.View(func(tx *bolt.Tx) error {
+		dbBucket := tx.Bucket([]byte(k8sRootBucket)).Bucket([]byte(DbBucket))
+		value := dbBucket.Get([]byte(key))
+		if value == nil {
+			return xerrors.Errorf("no outdated-api details for %s", key)
+		}
+		if err := json.Unmarshal(value, &outdatedapi); err != nil {
+			return xerrors.Errorf("failed to unmarshal JSON: %w", err)
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, xerrors.Errorf("failed to get the outdated-api %q: %w", key, err)
+	}
+	return outdatedapi, nil
+}
```

Comment:
can we use `outdatedapi`([types.OutDatedAPIData](https://github.com/aquasecurity/trivy-db/pull/233/files#diff-9aa20f1d30a7bc9bd5b07b0501cac0fe205afcb99c036c53d6a3bcf690ec4325R3)) for new databases?
If no - perhaps we need to return `value`(without unmarshal) and parse this into `Get() functions:
https://github.com/aquasecurity/trivy-db/blob/cf354a526ea44ec03e7a514249992a76ed76d792/pkg/k8s/api/outdated.go#L87-L96
if yes - I think we should choose a more "universal" name

Reply:
Not sure this one should be generic , it meant for outdated api datatype
if another k8s data will be added , we will have to add another file same as we do for vulns type we do not keep generic type for all.

Reply:
I am not sure that understood you correctly.
```
		if err := json.Unmarshal(value, &outdatedapi); err != nil {
			return xerrors.Errorf("failed to unmarshal JSON: %w", err)
		}
```
Will this work correctly with new k8s data?

Reply:
Yes ,as this file related to outdated api only, if a new k8d db will be added a new file should be introduced

Reply:
> as this file related to outdated api only

Then we need to move [this part](https://github.com/aquasecurity/trivy-db/blob/cf354a526ea44ec03e7a514249992a76ed76d792/pkg/db/k8sdetail.go#L58-L60) to [Get()](https://github.com/aquasecurity/trivy-db/blob/cf354a526ea44ec03e7a514249992a76ed76d792/pkg/k8s/api/outdated.go#L87-L96), because we will get error, when we try to unmarshal new data file(from `trivy-db-data`) with different structure.

Reply:
I see now what you mean I was redering to somthing else. Sorry for the confusion, ok I will dix it